archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,14 +23,41 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+
+        // Add a callback to update the cached primitive value when the property is changed.
+        this.prop.addCallback(new Runnable() {
+            @Override
+            public void run() {
+                primitiveValue = chooseValue();
+            }
+        });
     }
+
     /**
      * Get the current value from the underlying DynamicProperty
+     *
+     * @return
+     */
+    private boolean chooseValue() {
+        Boolean propValue = this.prop == null ? null : this.prop.getBoolean();
```

Comment:
I wasn't suggesting reverting the change to the get method, which is what your new test is effectively doing, but rather to put the old code from the get method into the new chooseValue method.  There it would be execute just once per property value change.